### PR TITLE
TYPE: Fix folding of doc comments

### PR DESCRIPTION
**Pull request by mchernyavsky**
Fixes https://github.com/intellij-rust/intellij-rust/issues/4058.

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/injected/RsDoctestLanguageInjector.kt [hunk 1]
```diff
@@ -145,7 +145,7 @@ class RsDoctestLanguageInjector : MultiHostInjector {
                 }
                 if (index == cratesEndIndex && !alreadyHasMain) {
                     if (prefix == null) prefix = StringBuilder()
-                    prefix.append("fn main() {")
+                    prefix.append("fn IntellijIdeaRulezzz() {")
```
Comment (Undin) [inline src/main/kotlin/org/rust/ide/injected/RsDoctestLanguageInjector.kt]: Is it intentional change?

Comment (mchernyavsky) [reply src/main/kotlin/org/rust/ide/injected/RsDoctestLanguageInjector.kt]: Yes, it disables folding in this case:
image

Comment (vlad20012) [reply src/main/kotlin/org/rust/ide/injected/RsDoctestLanguageInjector.kt]: Consider using another name to not confuse it with the one from completion, and also extract it to some constant.

Comment (mchernyavsky) [reply src/main/kotlin/org/rust/ide/injected/RsDoctestLanguageInjector.kt]: @vlad20012 Fixed.

**Pull request merged**